diff --git a/lib/vite/index.mjs b/lib/vite/index.mjs
--- a/lib/vite/index.mjs
+++ b/lib/vite/index.mjs
@@ -27571,7 +27571,8 @@
       ctx.isDevServer = config.command === "serve" && config.mode !== "production";
       ctx.isDevServerClientOnly = ctx.isDevServer && config.mode !== "ssr";
       await validatePlugin(ctx.opts);
-      mdxTransform = await createMdxTransformer(ctx);
+      const disableMdx = process.env.QWIK_CITY_DISABLE_MDX === "1";
+      mdxTransform = disableMdx ? async () => void 0 : await createMdxTransformer(ctx);
       qwikPlugin = config.plugins.find((p) => p.name === "vite-plugin-qwik");
       if (!qwikPlugin) {
         throw new Error("Missing vite-plugin-qwik");
diff --git a/lib/vite/index.cjs b/lib/vite/index.cjs
--- a/lib/vite/index.cjs
+++ b/lib/vite/index.cjs
@@ -27566,7 +27566,8 @@
       ctx.isDevServer = config.command === "serve" && config.mode !== "production";
       ctx.isDevServerClientOnly = ctx.isDevServer && config.mode !== "ssr";
       await validatePlugin(ctx.opts);
-      mdxTransform = await createMdxTransformer(ctx);
+      const disableMdx = "1" === process.env.QWIK_CITY_DISABLE_MDX;
+      mdxTransform = disableMdx ? async () => void 0 : await createMdxTransformer(ctx);
       qwikPlugin = config.plugins.find((p) => p.name === "vite-plugin-qwik");
       if (!qwikPlugin) {
         throw new Error("Missing vite-plugin-qwik");
